stats/stats_test.go

```diff
@@ -136,6 +136,76 @@ func TestZeroDivider(t *testing.T) {
 	}
 }
 
+func TestImportFunction1(t *testing.T) {
+	h := NewHistogram(0, 0.0001)
+	h.Record(0.04)
+	h.Record(0.05)
+	h.Record(0.08)
+	h.Record(0.10)
+	h.Record(0.50)
+	h.Print(os.Stdout, "TestImportFunction1", []float64{50})
+
+	var tests = []struct {
+		actual   Histogram
+		expected Histogram
+		msg      string
+	}{
+		{*h, *h.Export().Import(), ""},
+	}
+
+	for _, test := range tests {
+		if !reflect.DeepEqual(test.actual, test.expected) {
+			t.Errorf("%s: got %+v, not as expected %+v", test.msg, test.actual, test.expected)
+		}
+	}
+}
+
+func TestImportFunction2(t *testing.T) {
+	h := NewHistogram(0, 0.1)
+	h.Print(os.Stdout, "TestImportFunction2", []float64{50})
+
+	var tests = []struct {
+		actual   Histogram
+		expected Histogram
+		msg      string
+	}{
+		{*h, *h.Export().Import(), ""},
+	}
+
+	for _, test := range tests {
+		if !reflect.DeepEqual(test.actual, test.expected) {
+			t.Errorf("%s: got %+v, not as expected %+v", test.msg, test.actual, test.expected)
+		}
+	}
+}
+
+func TestImportFunction3(t *testing.T) {
```

Comment:
maybe one more test where you generate random points (and do the check in a loop, like 100 random histograms with 100 points in each)

Reply:
Hey Laurent,

So while writing this test case I realized there is a bug in my code. If all the values are mapped to one bucket (the JSON file contains one bucket), I sometimes incorrectly generate the original Histogram.

The reason being that in the Export function both the Start and End of the bucket are over-written by the min and max hence I lose the information about which slot in the histogram contains the actual count. This causes there to be an off-by-one error.

One solution might be to modify the Export function but I wanted to discuss this before I modified it.

Reply:
I don't get the jump between the values set to min and max (which helps making the interval more useful/increases the information available directly) and not being able to add that data correctly later ? if it was 1 bucket it'll be still one bucket... as you already export the offset and divisor, you can just make the same histogram and add the data back to it ?

Reply:
```
val := int32(math.Round((e.BucketMax - res.Offset) / res.Divider))
res.Hdata[indexSlice(histogramBucketValues, val)] = int32(bucket.Count)
```

To calculate the ```val``` I have to round because the ```histogramBucketValues``` contain exact integers. But when the bucket start or end value is replaced by the min or max this round might cause the data to be filled in the incorrect index (the next index).